Drag & Drop Support for Trackpad Devices

emreozcelik opened the pull request:
This pull request adds drag-and-drop support for trackpad devices and works well on my Samsung tablet.

I placed the code earlier for better organization, even though it would also work in the switch statement below.

By default, Samsung trackpads support normal drag-and-drop but don't allow double-tap followed by a quick drag. This is because the second tap is consumed as a move action.

The reason for this is that Samsung applies a threshold based on speed, time, and position before triggering an onTouch event. If the threshold is exceeded, it sends an onGenericMotion event, which is always treated as a move action.

However, double-tapping and then holding to drag works as expected, similar to regular drag-and-drop behavior.

emreozcelik committed:
Update Game.java

Press & Hold / Double-Tap & Hold for Selection or Drag & Drop Support for Trackpad Devices

ClassicOldSong commented:
Cool, but I think it's better to put them in the switch block. Also vibration may not be wanted especially when the tablet is put on table, maybe add a toggle for that or just remove the vibration?

emreozcelik committed:
Update for Drag & Drop

- Added an option under Input Settings to enable vibration feedback.
- Merged the code into the switch statement for better organization.

emreozcelik committed:
Update for Drag & Drop

- Added an option under Input Settings to set the drag-and-drop activation threshold duration.

emreozcelik commented:
Sure. I’ve moved the code into the switch block.

I’ve added an option to enable/disable vibration, with the default set to disabled, and also added an option to set the hold duration.

emreozcelik committed:
Update preferences.xml

Updated minimum value and step increments.

emreozcelik committed:
Update PreferenceConfiguration.java

Corrected the typo.

ClassicOldSong commented:
Great, I'll test it later on my tablets(not Samsung ones but have a trackpad). Thanks for your contribution!

ClassicOldSong committed:
Modify translation to specify the trackpad dnd action is Samsung only

ClassicOldSong commented:
Unfortunately it does not work on my device, it seems this behavior is Samsung specific. Merging now.
